File: split/scanrulespec.pas
function scanrulespec: halfword;
label
  21;
var
  q: halfword;
begin
  q := newrule;
  if curcmd = 35 then
    mem[q + 1].int := 26214
  else
  begin
    mem[q + 3].int := 26214;
    mem[q + 2].int := 0;
  end;
  21:
    if scankeyword(732) then
    begin
      scandimen(false, false, false);
      mem[q + 1].int := curval;
      goto 21;
    end;
  if scankeyword(733) then
  begin
    scandimen(false, false, false);
    mem[q + 3].int := curval;
    goto 21;
  end;
  if scankeyword(734) then
  begin
    scandimen(false, false, false);
    mem[q + 2].int := curval;
    goto 21;
  end;
  scanrulespec := q;
end;

File: cpp/src/scanrulespec.cpp
#include "scanrulespec.h"
#include "newrule.h"
#include "scankeyword.h"
#include "scandimen.h"

constexpr int default_rule = 26214; //0.4\thinspace pt


halfword scanrulespec(void)
{
	auto q = newrule();
	if (curcmd == vrule)
		width(q) = default_rule;
	else
	{
		height(q) = default_rule;
		depth(q) = 0;
	}
	while (true)
	{
		if (scankeyword("width"))
		{
			scan_normal_dimen();
			width(q) = curval;
			continue;
		}
		if (scankeyword("height"))
		{
			scan_normal_dimen();
			height(q) = curval;
			continue;
		}
		if (scankeyword("depth"))
		{
			scan_normal_dimen();
			depth(q) = curval;
			continue;
		}
		break;
	}
	return q;
}
